d2bs/kolbot/libs/common/Misc.js

```diff
@@ -2390,6 +2411,78 @@ CursorLoop:
 	removeListener: callback => removeEventListener('gamepacket', callback), // just a wrapper
 };
 
+/*
+
+new PacketBuilder() - create new packet object
+
+Example (Spoof 'reassign player' packet to client):
+    new PacketBuilder().byte(0x15).byte(0).dword(me.gid).word(x).word(y).byte(1).get();
+
+Example (Spoof 'player move' packet to server):
+    new PacketBuilder().byte(0x3).word(x).word(y).send();
+*/
+
+function PacketBuilder () {
+	/* globals DataView ArrayBuffer */
+	if (this.__proto__.constructor !== PacketBuilder) {
+		throw new Error("PacketBuilder must be called with 'new' operator!");
+	}
+
+	let pdata = [], dsize = 0;
+
+	let storeFields = (type, size) => (...args) => { // accepts any number of arguments
+		let nullTerm = 0;
+		let isString = false;
+
+		if (type === "StringZ") {
+			isString = true;
+			nullTerm = 1;
+			type = "String";
+		} else if (type === "String") {
+			isString = true;
+		}
+
+		for (let c = 0; c < args.length; c++) {
+			if (isString) {
+				size = args[c].length + nullTerm; // ArrayBuffers are initialized to 0, so just add 1 for null terminate
```

Comment:
consider removing isString?
```
		let nullTerm = -1;

 		if (type === "StringZ") {
			nullTerm = 1;
			type = "String";
		} else if (type === "String") {
			nullTerm = 0;
		}
 		for (let c = 0; c < args.length; c++) {
			if (nullTerm >= 0) {
				size = args[c].length + nullTerm; // ArrayBuffers are initialized to 0, so just add 1 for null terminate
```

Reply:
I use isString so it doesn't have to multiple of the same string comparison (found myself wanting to check type === "String" || type === "StringZ" a couple of times... in the unlikely event that 100 strings are passed this would be faster